skyreader/utils/handle_map_data.py

```diff
@@ -53,20 +59,35 @@ def extract_map(
             this_map = healpy.ud_grade(this_map, max_nside)
         mask = np.logical_and(~np.isnan(this_map), np.isfinite(this_map))
         equatorial_map[mask] = this_map[mask]
+        results_nside = result.result[f"nside-{nside}"]
 
-        for pixel_data in result.result[f"nside-{nside}"]:
+        for pixel_data in results_nside:
             pixel = pixel_data['index']
             value = pixel_data['llh']
-            if np.isfinite(value) and not np.isnan(value):
-                tmp_theta, tmp_phi = healpy.pix2ang(nside, pixel)
-                tmp_dec = np.pi/2 - tmp_theta
-                tmp_ra = tmp_phi
-                grid_map[(tmp_dec, tmp_ra)] = value
-                # nested_pixel = healpy.ang2pix(
-                #     nside, tmp_theta, tmp_phi, nest=True
-                # )
-                # uniq = 4*nside*nside + nested_pixel
-                # uniq_list.append(uniq)
+            nested_pixel = healpy.ring2nest(nside, pixel)
+            uniq = 4*nside*nside + nested_pixel
+            uniq_list.append(uniq)
+            tmp_theta, tmp_phi = healpy.pix2ang(nside, pixel)
+            tmp_dec = np.pi/2 - tmp_theta
+            tmp_ra = tmp_phi
+            grid_map[(tmp_dec, tmp_ra)] = value
+
+        # In case of pointed scans, it helps filling the first nside
+        # with empty pixels (especially for saving the multiorder map)
+        if nside == nsides[0]:
+            tot_npix = healpy.nside2npix(nside)
+            if tot_npix > len(results_nside):
+                ring_pixels = np.arange(tot_npix)
+                nest_pixels = healpy.ring2nest(nside, ring_pixels)
+                uniq_pixels = mhealpy.nest2uniq(nside, nest_pixels)
+                for uni, rin in zip(uniq_pixels, ring_pixels):
+                    if uni not in uniq_list:
+                        uniq_list.append(uni)
+                        tmp_theta, tmp_phi = healpy.pix2ang(nside, rin)
+                        tmp_dec = np.pi/2 - tmp_theta
+                        tmp_ra = tmp_phi
+                        grid_map[(tmp_dec, tmp_ra)] = np.nan
```

Comment:
This is specialized logic (hence all the indentations), so it's best placed in a helper function, something like `_fill_first_nside_empty()`.

Reply:
Implemented!